OMEdit/OMEditGUI/Search/SearchWidget.cpp

```diff
@@ -0,0 +1,400 @@
+/*
+ * This file is part of OpenModelica.
+ *
+ * Copyright (c) 1998-2014, Open Source Modelica Consortium (OSMC),
+ * c/o Linköpings universitet, Department of Computer and Information Science,
+ * SE-58183 Linköping, Sweden.
+ *
+ * All rights reserved.
+ *
+ * THIS PROGRAM IS PROVIDED UNDER THE TERMS OF GPL VERSION 3 LICENSE OR
+ * THIS OSMC PUBLIC LICENSE (OSMC-PL) VERSION 1.2.
+ * ANY USE, REPRODUCTION OR DISTRIBUTION OF THIS PROGRAM CONSTITUTES
+ * RECIPIENT'S ACCEPTANCE OF THE OSMC PUBLIC LICENSE OR THE GPL VERSION 3,
+ * ACCORDING TO RECIPIENTS CHOICE.
+ *
+ * The OpenModelica software and the Open Source Modelica
+ * Consortium (OSMC) Public License (OSMC-PL) are obtained
+ * from OSMC, either from the above address,
+ * from the URLs: http://www.ida.liu.se/projects/OpenModelica or
+ * http://www.openmodelica.org, and in the OpenModelica distribution.
+ * GNU version 3 is obtained from: http://www.gnu.org/copyleft/gpl.html.
+ *
+ * This program is distributed WITHOUT ANY WARRANTY; without
+ * even the implied warranty of  MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE, EXCEPT AS EXPRESSLY SET FORTH
+ * IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE CONDITIONS OF OSMC-PL.
+ *
+ * See the full OSMC Public License conditions for more details.
+ *
+ */
+/*
+ * @author Arunkumar Palansiamy <arunkumar.palanisamy@liu.se>
+ */
+
+#include "SearchWidget.h"
+#include "MainWindow.h"
+#include "Modeling/LibraryTreeWidget.h"
+#include "Util/Helper.h"
+#include "Options/OptionsDialog.h"
+#include <QtConcurrent/QtConcurrent>
```

Comment:
QtConcurrent/QtConcurrent header does not exist on Qt<5
This breaks the build on e.g. MacOS! Please fix!

Reply:
i don't have MAC OS to test it, but from the documentation i can see that Qt Concurrent features is available from Qt 4.4 http://doc.qt.io/qt-5/qtconcurrent.html

Reply:
OMEdit requires Qt 4.6.0 or newer and QtConcurrent is available since Qt 4.4 so it should work.
@sjoelund any idea?

Reply:
Does it require some module enabled in qmake?

Reply:
@hkiel it should work fine now after my fixes. At least hudson is not complaining anything.